fibsem/ui/FibsemMillingWidget.py

```diff
@@ -627,26 +646,88 @@ def _toggle_interactions(self, enabled: bool = True, caller: str = None, milling
         #     self.pushButton_run_milling.setStyleSheet("background-color: orange")
 
     def run_milling(self):
-
+        
         worker = self.run_milling_step()
         worker.finished.connect(self.run_milling_finished)
-        # worker.yielded.connect(self.update_milling_ui)
+        # worker.finished.connect(self.finish_progress_bar)
         worker.start()
+        
+
+    def start_progress_thread(self,params):
+
+        est_time = params[0]
+        idx = params[1]
+        total = params[2]
+
+        self.progressBar_milling.setVisible(True)
+        self.progressBar_milling.setValue(0)
+        self.progressBar_milling.setStyleSheet("QProgressBar::chunk "
+                          "{"
+                          "background-color: green;"
+                          "}")
+        self.progressBar_milling.setFormat(f"Milling stage {idx+1} of {total}: {est_time:.1f}s")
+
+        info = [idx, total, est_time]
+
+        self.progress_bar_worker = self.start_progress_bar(info)
+        self.progress_bar_worker.finished.connect(self.finish_progress_bar)
+
+        self.progress_bar_worker.start()
+
+    @thread_worker
+    def start_progress_bar(self,info):
```

Comment:
@Rohit-Kannachel  change info to a dictionary

Reply:
done